discovery/azure/azure.go

```diff
@@ -180,16 +190,31 @@ type azureResource struct {
 	ResourceGroup string
 }
 
+// virtualMachine represents an Azure virtual machine (which can also be created by a VMSS)
+type virtualMachine struct {
+	ID             *string
```

Comment:
Why are these all pointers?

Reply:
Actually I don't know why (this is a left over from the initial Azure support) I'm fine with changing this to non-pointer.

Reply:
This appears to be new code, so we may as well clean it up.

Reply:
Fixed in the latest commit